test/lib/pgBackRestTest/Env/Host/HostBackupTest.pm

```diff
@@ -1034,7 +1035,14 @@ sub configCreate
         $oParamHash{&CFGDEF_SECTION_GLOBAL}{cfgOptionName(CFGOPT_COMPRESS_LEVEL_NETWORK)} = 1;
     }
 
-    if (defined($$oParam{bCompress}) && !$$oParam{bCompress})
+    if (defined($oParam->{strCompressType}))
+    {
+        if ($oParam->{strCompressType} ne CFGOPTVAL_COMPRESS_TYPE_GZ)
+        {
+            $oParamHash{&CFGDEF_SECTION_GLOBAL}{cfgOptionName(CFGOPT_COMPRESS_TYPE)} = $oParam->{strCompressType};
```

Comment:
I'm a little confused by this - if defined($oParam->{strCompressType}) is true, couldn't it be 'none' (or later something like lz4) so here we're only setting the config file compress-type to 'none' or something other than 'gz'? But the if-else implies that strCompressType will not be set when 'none' since it default the CFGOPT_COMPRESS to 'n'.

Reply:
It can be set to 'none' if compress-type is explicitly set to none, as you can see in some expect logs.  The idea here is to have a few different ways to write the config to be sure we are handling them well.  So if `!defined(strCompressType)` then `compress=n` else if `defined` and `eq 'none'` then `compress-type=none`.

It's a bit convoluted but the alternative is to have two vars which I think would be worse.  One thing we could do is just stop using `compress` in our integration tests and depend on the unit tests to make sure compress/compress-type are resolved correctly.  I would be OK with that.

Reply:
Update to use only compress-type as we discussed.